cli/tools/run/mod.rs

```diff
@@ -60,13 +69,69 @@ To grant permissions, set them before the script argument. For example:
     deno_dir.upgrade_check_file_path(),
   );
 
-  let main_module = cli_options.resolve_main_module()?;
+  if flags.eszip {
```

Comment:
Can you please check that value in `cli/main.rs` in `run_subcommand` and if it's set create a completely new function (eg. `tools::run::run_eszip`). I would prefer we don't interleave this logic with current run subcommand.